specs/WebAuthenticationRequested.md

```diff
@@ -0,0 +1,280 @@
+
+```cpp
+    webview2->add_WebAuthenticationRequested(
+        Callback<ICoreWebView2WebAuthenticationRequestedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2WebAuthenticationRequestedEventArgs* args)
+            {
+                wil::com_ptr<ICoreWebView2Environment> webviewEnvironment;
+                m_appWindow->GetWebViewEnvironment()->QueryInterface(
+                    IID_PPV_ARGS(&webviewEnvironment));
+                wil::com_ptr<ICoreWebView2WebAuthenticationResponse> webAuthenticationResponse;
+                webviewEnvironment->CreateWebAuthenticationResponse(
+                    L"userName", L"password" , &webAuthenticationResponse);
+                args->put_Response(webAuthenticationResponse.get());
+
+                return S_OK;
+            })
+            .Get(),
+        &m_webAuthenticationRequestedToken));
+```
+
+```c#
+webView.CoreWebView2.WebAuthenticationRequested += delegate (object sender, CoreWebView2WebAuthenticationRequestedEventArgs args)
+{
+    CoreWebView2WebAuthenticationResponse response = _coreWebView2Environment.CreateWebAuthenticationResponse(
+        "User", "Pass");
+    args.Response = response;
+};
+```
+
+## Cancel authentication prompt
+The developer can block the authentication request. In this case, the default login dialog prompt will no longer be shown to the user and the server will respond as if the user clicked cancel.
+
+```cpp
+    webview2->add_WebAuthenticationRequested(
+        Callback<ICoreWebView2WebAuthenticationRequestedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2WebAuthenticationRequestedEventArgs* args)
+            {
+                args->put_Cancel(true);
+
+                return S_OK;
+            })
+            .Get(),
+        &m_webAuthenticationRequestedToken));
+```
+
+```c#
+webView.CoreWebView2.WebAuthenticationRequested += delegate (object sender, CoreWebView2WebAuthenticationRequestedEventArgs args)
+{
+    args.Cancel = true;
+};
+```
+
+## Read authorization challenge string
+Developer can read the authorization challenge string sent by server. Note that if the developer doesn't cancel or provide a response, the default login dialog prompt will be shown to the user.
+
+```cpp
+webview2->add_WebAuthenticationRequested(
+    Callback<ICoreWebView2WebAuthenticationRequestedEventHandler>(
+        [this](
+            ICoreWebView2* sender,
+            ICoreWebView2WebAuthenticationRequestedEventArgs* args)
+        {
+            args->get_Challenge(&challenge);
+            if (wcsncmp(challenge.get(), L"Expected login credentials") != 0)
```

Comment:
Is this a real thing you can do with a challenge? If this is more like up to the app and server to decide how they handle challenges, you could do a method like ValidateChallenge(challenge.get()) and a comment explaining its up to the app and server to decide on this